src/lib/components/Footer.svelte

```diff
@@ -2,37 +2,23 @@
 	import { X } from 'lucide-svelte';
 	import { base } from '$app/paths';
 	import { loadAttribute, saveAttribute } from '$lib/state/localstorage';
-	import { onMount } from 'svelte';
 	let dataPrivacyModal: HTMLDialogElement;
 	import { _, locale } from 'svelte-i18n';
+	import { DEFAULT_LOCALE, KEY_LOCALE, Locale } from '$lib/utility/locale';
 
 	// eslint-disable-next-line no-undef
 	const APP_NAME = __APP_NAME__;
 	// eslint-disable-next-line no-undef
 	const APP_VERSION = __APP_VERSION__;
 
-	const Locale = {
-		German: 'de',
-		English: 'en'
-	};
-	const KEY_LOCALE: string = 'locale';
-	const DEFAULT_LOCALE: string = Locale.German;
-	let selected_locale: string = $state(DEFAULT_LOCALE);
+	let selected_locale: string = $state(loadAttribute<string>(KEY_LOCALE) || DEFAULT_LOCALE);
 	let isModalOpen: boolean = $state(false);
 
-	onMount(() => {
-		const locale = loadAttribute<string>(KEY_LOCALE);
-		selected_locale = locale || DEFAULT_LOCALE;
-	});
-
-	$effect(() => {
-		$locale = selected_locale || DEFAULT_LOCALE;
-	});
-
 	function toggleLanguage() {
 		selected_locale = selected_locale === Locale.German ? Locale.English : Locale.German;
 		saveAttribute(KEY_LOCALE, selected_locale);
-		document.title = $_('app_title', { locale: selected_locale });
+		$locale = selected_locale;
```

Comment:
Instead of having the locale be set inside the `$effect` execute it here. Makes it more clearer when locales are changed.